x/crosschain/keeper/keeper_cross_chain_tx_vote_outbound_tx.go

```diff
@@ -160,17 +161,27 @@ func (k msgServer) VoteOnObservedOutboundTx(goCtx context.Context, msg *types.Ms
 			} else {
 				switch oldStatus {
 				case types.CctxStatus_PendingOutbound:
+
+					gasLimit, err := k.GetRevertGasLimit(ctx, cctx)
+					if err != nil {
+						return errors.New("can't get revert tx gas limit" + err.Error())
+					}
+					if gasLimit == 0 {
+						// use same gas limit of outbound as a fallback -- should not happen
+						gasLimit = cctx.OutboundTxParams[0].OutboundTxGasLimit
+					}
+
 					// create new OutboundTxParams for the revert
-					cctx.OutboundTxParams = append(cctx.OutboundTxParams, &types.OutboundTxParams{
-						Receiver:        cctx.InboundTxParams.Sender,
-						ReceiverChainId: cctx.InboundTxParams.SenderChainId,
-						Amount:          cctx.InboundTxParams.Amount,
-						CoinType:        cctx.InboundTxParams.CoinType,
-						// NOTE(pwu): revert gas limit = initial outbound gas limit set by user
-						//TODO: determine a specific revert gas limit https://github.com/zeta-chain/node/issues/1065
-						OutboundTxGasLimit: cctx.OutboundTxParams[0].OutboundTxGasLimit,
-					})
-					err := k.PayGasAndUpdateCctx(
```

Comment:
does this remove the payment of gas?

Reply:
What do you mean? The function is still called.

Reply:
oh yeah my bad missed that.